Bustracker-Adminpanel implementation

Gajendra204 opened the pull request:
Bustracker-Adminpanel implementation

Gajendra204 committed:
Code review

boring-anime commented on src/api.ts:
remove commented code

boring-anime commented on README.md:
The README file should include information about the project and its dependencies. and guidelines to set up the project

boring-anime reviewed (changes requested):
It dont have any test cases

boring-anime commented on src/api/auth.ts:
```diff
@@ -0,0 +1,20 @@
+export const loginAdmin = async (data: LoginData) => {
+  try {
+    const response = await axiosInstance.post("/auth/admin/login", data);
+    const token = response.data.data.token;    
```
Accessing response.data.data.token assumes the response always contains data.data.token. If the API changes or fails, this will throw. Add a check or fallback for missing properties.

boring-anime commented on src/api/auth.ts:
```diff
@@ -0,0 +1,20 @@
+    localStorage.setItem("busTrackerAdminToken", token); 
+    return response.data;
+  } catch (error) {
+    console.error("Login error:", error);
```
Right now the code uses console.error for logging. This will only show messages in the console if someone is actively watching, which isn’t practical in production. How will you track or review these errors later? Consider using a proper logging utility or monitoring service so that logs can be stored, searched, and monitored effectively.

boring-anime commented on src/api/auth.ts:
```diff
@@ -0,0 +1,20 @@
+  }
+};
+
+export const registerAdmin = async (data: RegisterData) => {
```
The registerAdmin function does not handle errors. Wrap the request in a try-catch block to handle failed registration attempts consistently with loginAdmin.

boring-anime commented on src/App.css:
Why does it have a blank file?

boring-anime commented on src/api/buses.ts:
The functions lack a return type annotation. Explicitly specifying the return type improves type safety and code readability.

boring-anime commented on src/api/drivers.ts:
```diff
@@ -0,0 +1,21 @@
+  return response.data;
+};
+
+export const createDriver = async (data: any) => {
```
The parameter data in createDriver is typed as any. Consider defining and using a proper TypeScript interface for driver data to improve type safety.

boring-anime commented on src/api/drivers.ts:
```diff
@@ -0,0 +1,21 @@
+  return response.data;
+};
+
+export const updateDriver = async (id: string, data: { name: string; phone: string }) => {
```
The parameter data in updateDriver is explicitly typed, which is good. If the driver object shape is reused elsewhere, consider extracting it into a shared interface/type.

boring-anime commented on src/api/types.ts:
```diff
@@ -0,0 +1,96 @@
+  password: string;
+}
+
+export type AdminProfile = {
```
The AdminProfile type is declared as a type alias, while similar structures are declared as interfaces. For consistency, consider using interface for all object shapes unless you need type alias features.

Gajendra204 committed:
Adressed PR comments: removed commented code and update the README file

Gajendra204 committed:
Adressed PR comments: Implemeted sentry service for error logging and try-catch block

Gajendra204 committed:
Refactor: Improve type safety and consistency

- Explicitly typed function return values.
- Replaced 'any' with a proper interface for driver data.
- Standardized object shapes to use interfaces.

Gajendra204 committed:
fix(api): updated buses, drivers, and students API functions to return response.data.data for correct data structure

Gajendra204 committed:
fix: remove redundant .data access in react-query hooks

Gajendra204 committed:
test: add unit tests for hooks and context (Auth, Buses, Drivers, Routes, Students, PrivateRoute)

- Added unit tests for AuthContext and useAuth hook
- Added unit tests for PrivateRoute component
- Added unit tests for useBuses, useDrivers, useRoutes, and useStudents hooks
- Setup and verified Jest + React Testing Library configuration

Gajendra204 committed:
Removed Parents component and page

Gajendra204 committed:
Deleted Profile Component

Gajendra204 committed:
refactor: Create utility-based style files for ManagementTable, ManagementFormModal, Sidebar, and RouteForm components

Gajendra204 committed:
refactor(ui): Extract utility-based styles and enhance design for ManagementTable, ManagementFormModal, Sidebar, and RouteForm

Gajendra204 committed:
refactor(api): convert arrow functions to function declarations in API's service files

Gajendra204 committed:
changed the students count to dynamic

Gajendra204 committed:
refactor(styling): implement consistent styling for route components

Gajendra204 committed:
Refactor: Refactor the structure

Gajendra204 committed:
Refactor: Deleted unnecessary nested folder

Gajendra204 committed:
Refacter UI
